tests/phpunit/tests/date/currentTime.php

```diff
@@ -157,4 +157,36 @@ public function test_should_work_with_deprecated_timezone() {
 		$this->assertSame( gmdate( $format ), $current_time_gmt, 'The dates should be equal [3]' );
 		$this->assertSame( $datetime->format( $format ), $current_time, 'The dates should be equal [4]' );
 	}
+
+	/**
+	 * @ticket 57998
+	 *
+	 * @dataProvider data_partial_hour_timezones_with_timestamp
+	 *
+	 * @param float $partial_hour Partial hour GMT offset to test.
+	 */
+	public function test_partial_hour_timezones_with_timestamp( $partial_hour ) {
+		update_option( 'gmt_offset', $partial_hour );
```

Comment:
This might also need timezone_string option to be reset (deleted?), since it overrides gmt_offset via filter.